playerctl/playerctl-daemon.c

```diff
@@ -332,9 +332,40 @@ static void context_remove_player(struct PlayerctldContext *ctx, struct Player *
     }
 }
 
+/**
+ * Returns the newly activated player
+ */
+static struct Player *context_shift_active_player(struct PlayerctldContext *ctx) {
+    GError *error = NULL;
+    struct Player *p;
+
+    if (!(p = context_get_active_player(ctx))) {
+        return NULL;
+    }
+    context_remove_player(ctx, p);
+    context_add_player(ctx, p);
+    p = context_get_active_player(ctx);
+
+    context_emit_active_player_changed(ctx, &error);
+    if (error != NULL) {
+        g_warning("could not emit active player change: %s", error->message);
+        g_clear_error(&error);
+    }
+    player_update_position_sync(p, ctx, &error);
```

Comment:
Update the position before emitting signals or it sends the old position.

Reply:
See https://github.com/altdesktop/playerctl/pull/173/commits/8ece9efd5ce0a6ab0f14f4cfd1e194d4736fa76e